Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -237,6 +265,60 @@ public AndConstraint<XDocumentAssertions> NotBeEquivalentTo(XDocument unexpected
             return new AndWhichConstraint<XDocumentAssertions, XElement>(this, xElement);
         }
 
+        /// <summary>
+        /// Asserts that the <see cref="XDocument.Root"/> element of the current <see cref="XDocument"/> has the specified occurrence of
+        /// child elements with the specified <paramref name="expected"/> name.
+        /// </summary>
+        /// <param name="expected">
+        /// The full name <see cref="XName"/> of the expected child element of the current document's <see cref="XDocument.Root"/> element.
+        /// </param>
+        /// <param name="occurrenceConstraint">
+        /// A constraint specifying the number of times the specified elements should appear.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndWhichConstraint<XDocumentAssertions, XElement> HaveElement(XName expected,
+            OccurrenceConstraint occurrenceConstraint, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(expected, nameof(expected),
+                    "Cannot assert the document has an element count if the element name is <null>.");
+
+            bool success = Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Cannot assert the count if the document itself is <null>.");
+
+            IEnumerable<XElement> xElements = Enumerable.Empty<XElement>();
+
+            if (success)
+            {
+                Execute.Assertion
+                .ForCondition(Subject.Root is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    "Expected {context:subject} to have root element with child {0}{reason}, but it has no root element.",
+                    expected.ToString());
+
+                xElements = Subject.Root.Elements(expected);
+                int actualCount = xElements.Count();
+
+                Execute.Assertion
+                    .ForConstraint(occurrenceConstraint, actualCount)
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith(
```

Comment:
You'll need write this a bit differently to get the "at least"/"at most"/etc included in the failure message.
https://github.com/fluentassertions/fluentassertions/blob/b7d63c02cf705f13aef6f568c84f29c615596455/Src/FluentAssertions/Primitives/StringAssertions.cs#L941-L946

Reply:
Ahh.. thats why the `Mode` property exists :D

Reply:
How do I get the "at least" not be quoted?

`Did not expect any exception, but found Xunit.Sdk.XunitException with message "Expected element to have "at least" 2 child element(s) "child", but found "child"."`

Reply:
Ahh got it...